### PR TITLE
Fix issue 3298

### PR DESCRIPTION
**Pull request by romainfrancois**
This is about issue #3298. Making it illegal for the rhs of a `mutate` to be a data frame: 

```
> df <- data.frame("a" = c(1, 2, 3), "b" = c(2, 3, 4), "base_col" = c(3, 4, 5))
> mutate(df, new_col = data.frame(1:3) )
 Show Traceback
 
 Rerun with Debug
 Error in mutate_impl(.data, dots) : 
  Column `new_col` is of unsupported class data.frame 
```

- In the discussion #3298 @krlmlr mentioned that this should be a warning first, I don't see why
- I'm not sure about the error message

### CODE REVIEW

**Comment by krlmlr** [comment]
We want a bugfix-only release, this is a (very convoluted) API change. The error-throwing behavior of a function changes, and this might trigger failures in revdepchecks.

**Comment by romainfrancois** [comment]
Sure. well the PR is there for when we want it. I think your assessment is a bit strong about the convolutiveness. I'd qualify this as a bug ...

File: tests/testthat/test-mutate.r [hunk 1]
```diff
@@ -475,30 +475,23 @@ test_that("mutate works on empty data frames (#1142)", {
   expect_equal(length(res), 1L)
 })
 
-test_that("mutate handles 0 rows rowwise (#1300)", {
```
Comment (krlmlr) [inline tests/testthat/test-mutate.r]: Why is this test now removed?

Comment (romainfrancois) [reply tests/testthat/test-mutate.r]: I guess the first expectation still makes sense. The two following ones are replaced by the new tests. 

```
> a <- data_frame(x = 1)
> b <- data_frame(y = character())
> 
> g <- function(y) {
+     1
+ }
> f <- function() {
+     b %>% rowwise() %>% mutate(z = g(y))
+ }
> 
> res <- f()
> expect_equal(nrow(res), 0L)
> 
> expect_error(
+     a %>% mutate(b = f()),
+     "Column `b` must be length 1 (the number of rows), not 2",
+     fixed = TRUE
+ )
Error: `a %>% mutate(b = f())` threw an error with unexpected message.
Expected match: "Column `b` must be length 1 (the number of rows), not 2"
Actual message: "Column `b` is of unsupported class data.frame"
> 
> expect_error(
+     a %>% rowwise() %>% mutate(b = f()),
+     "Column `b` must be length 1 (the group size), not 2",
+     fixed = TRUE
+ )
Error: `a %>% rowwise() %>% mutate(b = f())` threw an error with unexpected message.
Expected match: "Column `b` must be length 1 (the group size), not 2"
Actual message: "Column `b` is of unsupported class data.frame"
```

**Comment by romainfrancois** [comment]
Sure, no trouble. Please let me know what procedure to follow then.

**Comment by krlmlr** [comment]
It's `styler::style_pkg(strict = FALSE)`, you can also do `styler::style_file(..., strict = FALSE)` to format just one file.

**Comment by romainfrancois** [comment]
`styler::style_pkg(strict = FALSE)` wants to change many files. 

capture d ecran 2018-03-06 a 22 13 38

Many of which I did not touch. Happy to run it still, perhaps outside of these branches though.

**Comment by krlmlr** [comment]
Sorry, I forgot we haven't applied styler to the codebase yet (and we probably want the dev version). Can you please revert the last commit and perhaps keep only the changes that are related to this PR?

**Pull request merged**